### PR TITLE
Re-write Rank component from scratch using TDD

### PR DESCRIPTION
**Pull request by adrianclay**

### CODE REVIEW

File: components/Rank/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import {mount} from 'enzyme'
+import React from 'react'
+import Rank from '../'
+
+describe('Rank', () => {
+  beforeEach(() => {
+    console.error = jest.fn((error) => {
```
Comment (adrianclay) [inline components/Rank/__tests__/index.js:7]: I'm not happy with this overwriting.  I imagine that ` console.error ` will remain mocked.  Moving it to a global test bootstrap would be 1 option.  Good to hear peoples opinions of this.

File: components/Rank/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  it('displays second place', () => {
+    expect(mount(<Rank ranking={2}/>).text()).toBe("2")
+  })
+
+  it('rejects missing ranking', () => {
+    expect(() => mount(<Rank/>)).toThrow()
+  })
+
+  it('rejects non-stringy rankings', () => {
+    expect(() => mount(<Rank ranking="dodgy"/>)).toThrow()
```
Comment (camillavk) [inline components/Rank/__tests__/index.js:25]: If you're going to test that it throws an error for missing or wrong rankings, would it maybe also be worth testing what error it throws? I.e. would these two situations throw different errors? I didn't go down this route personally but wondering if you do think it is something worth testing that maybe it needs to be even clearer? (or am I talking nonsense?)

File: components/Rank/__tests__/index.js (hunk 1 above)
Comment (rrpff) [reply components/Rank/__tests__/index.js:7]: My thoughts on this:

We could use Sinon to stub and restore the function:

```js
// stub
sinon.stub(console, 'error', warning => { throw new Error(warning) })

// restore
console.error.restore()
```

We could use Jest [Setup and Teardown](https://facebook.github.io/jest/docs/setup-teardown.html) functions to run this before and after each test.

Additionally, if we wanted to specify this on a global level, we could put it in a global setup file, such as `test/setup.js`, and require it before each test by putting this in `package.json`:

```json
"jest": {
  "setupFiles": [
    "./test/setup.js"
  ]
}
```

The downside of this is any test in the suite will fail if any `error` is logged. This demands that all dependencies use `console.error` semantically, which many not be the case.

Comment (adrianclay) [reply components/Rank/__tests__/index.js:7]: I couldn't get the global level version working, 😢 

```console
    ReferenceError: beforeAll is not defined
```

File: components/Rank/__tests__/index.js (hunk 2 above)
Comment (adrianclay) [reply components/Rank/__tests__/index.js:25]: I'm not sure how much value there is in asserting the propTypes have been populated.  I think that stems from an unresolved question in my mind as to whether the types are are adding value.

Comment (adrianclay) [reply components/Rank/__tests__/index.js:25]: I only wrote those tests to get into a state which forced me to write the propTypes.

Comment (craigjbass) [reply components/Rank/__tests__/index.js:25]: Isn't the accepted answer here to use Flow?

**Pull request closed without merging**